docs/rfcs/no-unusable-components-line-in-the-sand.md

```diff
@@ -0,0 +1,75 @@
+# No unusable Source components
+
+This proposes a line in the sand regarding how Source components should behave when their ideal version requires technologies that a client may not offer.
+
+## Example problem
+
+A toggle switch is similar to a checkbox, but unlike a checkbox, the state it sets should apply immediately (no submit button).
+
+It needs javascript to achieve this. There is no obvious default widget to fall back to – it's not just a visual treatment of standard widget.
+
+If we server render it, users without javascript enabled (or who have not yet downloaded the bundle) will be presented with an unresponsive CTA.
+
+## Proposal
+
+A new line in the sand for Source: 
+
+> Do not present a user with a component they cannot use.
+
+Given the example above, this could mean either:
+
+- not displaying it to users without javascript
+- identifying and displaying a non-JS fallback
+
+Note that this *does not mandate fallbacks*. A team could decide that the component is an enhancement and not critical, and so they decide to just not show it to people without javascript.
```

Comment:
I'm unsure about this line, apologies if I'm missing something. Is the suggestion:

> We (Design System Team) don't need to create fallbacks for every component, because teams might not need to use the fallback. We only create a fallback when the component has a known critical role in a project that consumes Source.

?

Reply:
It feels like we need to define standard vs enhanced and the various levels of support, e.g.:

1. Not shown – a component is not shown if it cannot be used in the environment
2. Standard – a minimally functional but stable component that works everywhere
3. Enhanced – a streamlined and delightful experience for modern browsers and environments

Reply:
i think i'm suggesting that Source components should provide mechanism for fallbacks where they are needed, and consumer can choose whether they use them?

but i agree it's not clear – i think i've confused what should be in source, and what we could do elsewhere. i'll look at that again